src/main/webapp/app/exercises/shared/exam-exercise-row-buttons/exam-exercise-row-buttons.component.html

```diff
@@ -141,19 +141,32 @@
         </button>
     }
     @if (course.isAtLeastInstructor && exercise.type === exerciseType.PROGRAMMING) {
-        <button
-            jhiDeleteButton
-            [entityTitle]="exercise.title || ''"
-            deleteQuestion="artemisApp.programmingExercise.delete.question"
-            (delete)="deleteProgrammingExercise($event)"
-            [dialogError]="dialogError$"
-            deleteConfirmationText="artemisApp.exercise.delete.typeNameToConfirm"
-            [additionalChecks]="{
-                deleteStudentReposBuildPlans: 'artemisApp.programmingExercise.delete.studentReposBuildPlans',
-                deleteBaseReposBuildPlans: 'artemisApp.programmingExercise.delete.baseReposBuildPlans'
-            }"
-        >
-            <fa-icon [icon]="faTrash" />
-        </button>
+        @if (localVCEnabled) {
+            <button
+                jhiDeleteButton
+                [entityTitle]="exercise.title || ''"
+                deleteQuestion="artemisApp.programmingExercise.delete.questionLocalVC.single"
+                (delete)="deleteProgrammingExercise($event)"
+                [dialogError]="dialogError$"
+                deleteConfirmationText="artemisApp.exercise.delete.typeNameToConfirm"
+            >
+                <fa-icon [icon]="faTrash" />
+            </button>
+        } @else {
+            <button
+                jhiDeleteButton
+                [entityTitle]="exercise.title || ''"
+                deleteQuestion="artemisApp.programmingExercise.delete.question"
+                (delete)="deleteProgrammingExercise($event)"
+                [dialogError]="dialogError$"
+                deleteConfirmationText="artemisApp.exercise.delete.typeNameToConfirm"
+                [additionalChecks]="{
+                    deleteStudentReposBuildPlans: 'artemisApp.programmingExercise.delete.studentReposBuildPlans',
+                    deleteBaseReposBuildPlans: 'artemisApp.programmingExercise.delete.baseReposBuildPlans'
+                }"
+            >
+                <fa-icon [icon]="faTrash" />
+            </button>
+        }
```

Comment:
These buttons seem to have many things in common - perhaps using the ternary operator in `deleteQuestion` and `additionalChecks` would make this template shorter and reduce duplication. You could then get rid of the `@if...@else` construct.

What do you think? There are multiple instances of this.

Reply:
done